jscomp/ml/ast_untagged_variants.ml

```diff
@@ -189,7 +189,7 @@ let get_block_type ~env (cstr : Types.constructor_declaration) :
     block_type option =
   match (process_untagged cstr.cd_attributes, cstr.cd_args) with
   | false, _ -> None
-  | true, Cstr_tuple [{desc = Tconstr _} as t] when get_block_type_from_typ ~env t |> Option.is_some -> get_block_type_from_typ ~env t
```

Comment:
This does not work for e.g. `Tarrow`.
@zth any special reason for having it there?

Reply:
No that has looks like an oversight from me.